AMP-118975 sendBeacon() warning (#453)
amplitude/amplitude-docs: content/collections/browser_sdk/en/browser-sdk-2.md

Unlike standard network requests, sendBeacon sends events in the background, even if the user closes the browser or leaves the page.

{{partial:admonition type="warning" heading=""}}

> `sendBeacon` sends events in the background. As a result, events `sendBeacon` dispatches don't return a server response. Keep the following in mind if you use `sendBeacon`:

Reviewer: Nit: Flows a bit better (subjective) and be consistent with plurality (events, dispatches, a server response). 
```suggestion
`sendBeacon` sends events in the background. As a result, events dispatched by `sendBeacon` don't return server responses. Keep the following in mind if you use `sendBeacon`:
```
or 
```suggestion
`sendBeacon` sends events in the background. As a result, events `sendBeacon` dispatches don't return server responses. Keep the following in mind if you use `sendBeacon`:```

> 1. There is no retry, even for 4xx or 5xx responses. As a result, you may lose events.

Reviewer: Be explicit with what is not being retried
```suggestion
1. Requests are not retried, including failed requests with 4xx or 5xx responses, so events may be lost.
```

> 1. You might end up losing events becase there is no retry, even when 4xx or 5xx responses occur.

Reviewer 2: ```suggestion
1. There is no retry, even for 4xx or 5xx responses. As a result, you may lose events.
```

> 2. Unlike `fetch`, for which the SDK waits for a response before it sends another request, `sendBeacon` can send events in parallel. This can result in the server receiving events out of order, and lead to some UTM properties not being set for session start events for example.

Reviewer: Consider putting information for `sendBeacon` first, with alternative details (`fetch`) after, as this is what we want the user to keep in mind.
```suggestion
2. Event order cannot be guaranteed, as `sendBeacon` may send events in parallel. This can lead to some UTM properties not being set, e.g. for session start events. In contrast, while using `fetch`, the SDK waits for responses before proceeding, guaranteeing event order.
```

> 2. Event order cannot be guaranteed, as `sendBeacon` may send events in parallel. This can lead to some UTM properties not being set, e.g. for session start events. In contrast, while using `fetch`, the SDK waits for responses before proceeding, guaranteeing event order.

Reviewer: Looks like my suggestion of using `e.g.` got flagged by a lint rule. Hopefully this captures the intent without being overly verbose.
```suggestion
2. Event order cannot be guaranteed, as `sendBeacon` may send events in parallel. This can lead to concurrency issues. For example, session start events that set UTM properties could be sent after other events in the session, resulting in those events lacking UTM properties. In contrast, while using `fetch`, the SDK waits for responses before proceeding, guaranteeing event order.
```

Author: I will just replace e.g. with for example. Session start events don't set UTMs. Identify events set UTMs.

> 2. Unlike `fetch`, which returns a response that the SDK waits for before sending another request to guarantee event order received by the server, `sendBeacon` may send events in parallel, potentially resulting in the server receiving events out of order. This can lead to some UTM properties not being set for session start events for example.

Reviewer 2: ```suggestion
2. Unlike `fetch`, for which the SDK waits for a response before it sends another request, `sendBeacon` can send events in parallel. This can result in the server receiving events out of order, and prevent some UTM properties from being used as session start events.
```

Author: Hi @markzegarelli, thanks for reviewing it. I'd like to keep the word "set" as the server actually sets user properties based on a table. How about

```
`sendBeacon` can send events in parallel. This can result in the server receiving events out of order, and lead to some UTM properties not being set for session start events for example.
```
